model/task_history.go

```diff
@@ -351,16 +354,9 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 	return chunk, nil
 }
 
-func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string, error) {
-	session, mdb, err := db.GetGlobalSessionFactory().GetSession()
-	if err != nil {
-		return nil, errors.Wrap(err, "problem getting database session")
-	}
-	defer session.Close()
-
-	session.SetSocketTimeout(time.Minute)
-
-	pipeline := mdb.C(task.Collection).Pipe(
+func (self *taskHistoryIterator) GetDistinctTestNames(env evergreen.Environment, ctx context.Context, numCommits int) ([]string, error) {
+	opts := options.Aggregate().SetBatchSize(0).SetMaxTime(time.Minute)
```

Comment:
Is it still necessary to set MaxTime if we're going to kill the cursor when the context gets killed?

Reply:
I was thinking of that case where the app server got mildly taken down by someone using this page, and when they closed the page it got better. This would prevent us from being reliant on the user to close the page. We could set the timeout later though, if preferred.